api/webhook.go

```diff
@@ -257,19 +256,9 @@ func PostWebhook(c *gin.Context) {
 		return
 	}
 
-	if b.GetEvent() == constants.EventComment {
-		number, err := parsePRNumberFromRef(b.GetRef())
-		if err != nil {
-			retErr := fmt.Errorf("%s: failed to get pull_request number for %s: %v", baseErr, r.GetFullName(), err)
-			util.HandleError(c, http.StatusInternalServerError, retErr)
-
-			h.SetStatus(constants.StatusFailure)
-			h.SetError(retErr.Error())
-
-			return
-		}
-
-		commit, branch, baseref, err := source.FromContext(c).GetPullRequest(u, r, number)
+	// if this is a comment on a pull_request event
+	if strings.EqualFold(b.GetEvent(), constants.EventComment) && webhook.PRNumber != 0 {
```

Comment:
We are now passing through pull request number on the `webhook` struct and are using that instead of calling the utility function to strip it from the ref.